src/commands/Changelog.ts

```diff
@@ -0,0 +1,318 @@
+                            EOL
+                    );
+                }
+            }
+
+            writeStream.write(EOL + "Removed syntax nodes:" + EOL + EOL);
+            writeStream.write(
+                "Name:" +
+                    csvSeparator +
+                    "Removed from language(s):" +
+                    csvSeparator +
+                    "Used for calculating the metric(s):" +
+                    csvSeparator +
+                    "Was explicitly only activated for language(s):" +
+                    EOL
+            );
+
+            for (const entry of this.getAll().values()) {
+                if (entry.remainingLanguages.size + entry.addedLanguages.size === 0) {
+                    const mapping = metricMappings.get(entry.expression);
+                    if (mapping !== undefined) {
+                        const activatedForLangOutput =
+                            mapping.activated_for_languages === undefined
+                                ? ""
+                                : mapping.activated_for_languages;
+                        writeStream.write(
+                            escapeForCsv(entry.expression) +
+                                csvSeparator +
+                                Array.from(entry.removedLanguages) +
+                                csvSeparator +
+                                mapping.metrics +
+                                csvSeparator +
+                                activatedForLangOutput +
+                                EOL
+                        );
+                    } else {
+                        throw new Error(
+                            "Programming mistake: No existing expression mapping for a syntax node that is not new: " +
+                                entry.expression
+                        );
+                    }
+                }
+            }
+
+            writeStream.write(
+                EOL +
+                    "Already known and still used syntax nodes which were removed from or added to some language(s):" +
+                    EOL +
+                    EOL
+            );
+            writeStream.write(
+                "Name:" +
+                    csvSeparator +
+                    "Added to language(s):" +
+                    csvSeparator +
+                    "Removed from language(s):" +
+                    csvSeparator +
+                    "Remains in language(s):" +
+                    csvSeparator +
+                    "Used for calculating the metric(s):" +
+                    csvSeparator +
+                    "Was explicitly only activated for language(s):" +
+                    EOL
+            );
+
+            for (const entry of this.getAll().values()) {
+                if (
+                    entry.remainingLanguages.size > 0 ||
+                    (entry.addedLanguages.size > 0 && entry.removedLanguages.size > 0)
+                ) {
```

Comment:
Let's create a method to give this condition a readable name like `hasExpressionChanged()`. Is it correct to have remainingLanguages > 0 as an OR condition?

Reply:
Done

Reply:
It should be correct. Either there are remaining languages, in which case its clear that this is a modified node type and no completely removed or new one, or there are both added and removed languages, for the special case that every old language was removed but there were new languages added.

However, I changed that now to a more intuitive condition by using the filed `isNewNode`.